src/components/header/search/search.jsx

```diff
@@ -1,33 +1,16 @@
-import React, { useEffect } from 'react'
+import React from 'react'
 import PropTypes from 'prop-types'
-
-import SearchIcon from './search-icon.svg'
+import '@docsearch/css'
 import * as style from './search.module.css'
-
-import { initialiseAlgolia } from './algolia-search'
-
-const Search = ({ background }) => {
-  useEffect(() => {
-    initialiseAlgolia()
-  }, [])
-
+import { DocSearch } from '@docsearch/react'
+const Search = () => {
   return (
-    <div
-      className={`${style.search} ${
-        background === 'solid'
-          ? style.backgroundSolid
-          : style.backgroundTransparent
-      }`}
-    >
-      <input
-        id='search'
+    <div className={`${style.search}`}>
+      <DocSearch
+        appId='BH4D9OD16A'
+        apiKey='be16e460d9d03fa711df82d525dec3c1'
+        indexName='nubots'
```

Comment:
Perhaps something like this will fix the problem with search result URLs pointing to the main site (instead of localhost/deploy preview):

```jsx
<DocSearch
  // ... other props
  transformItems={(items) => {
    return items.map((item) => {
      // Get the origin (domain + possible port) from the result link
      const { origin } = new URL(item.url);
      return {
        ...item,
        // Replace the origin in the result URL with our current origin,
        // so the links work for local development and deploy previews
        url: item.url.replace(origin, window.location.origin),
      }
    });
  }}
/>
```

It's using the [`transformItems`](https://docsearch.algolia.com/docs/api#transformitems) prop to update the URLs on the search results before showing them.

Reply:
That did the trick, nice find!